packages/monaco/src/browser/monaco-status-bar-contribution.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from 'inversify';
+import { DisposableCollection } from '@theia/core';
+import { FrontendApplicationContribution, FrontendApplication, StatusBar, StatusBarAlignment } from "@theia/core/lib/browser";
+import { EditorCommands, EditorManager, EditorWidget } from '@theia/editor/lib/browser';
+import { OutlineViewService } from '@theia/outline-view/lib/browser/outline-view-service';
+import { MonacoEditor } from './monaco-editor';
+
+@injectable()
+export class MonacoStatusBarContribution implements FrontendApplicationContribution {
+
+    protected readonly toDispose = new DisposableCollection();
+
+    constructor(
+        @inject(OutlineViewService) protected readonly outlineViewService: OutlineViewService,
+        @inject(EditorManager) protected readonly editorManager: EditorManager,
+        @inject(StatusBar) protected readonly statusBar: StatusBar
+    ) { }
+
+    onStart(app: FrontendApplication): void {
+        this.statusBar.onStatusBarEntriesLoaded(() => {
+            this.removeConfigTabSizeWidget();
+        });
+
+        this.editorManager.onCurrentEditorChanged(widget => {
+            if (widget) {
+                this.setConfigTabSizeWidget();
+
+                const editor = this.editorManager.currentEditor;
+                if (editor && this.getModel(editor)) {
+                    const editorModel = this.getModel(editor);
+                    if (editorModel) {
```

Comment:
this check also won't help, it will throw an exception in `this.getModel(editor)` already in this case

Reply:
because we are going to change the return type of `getModel()` to `monaco.editor.IModel | undefined`, the `getModel()` function wouldn't throw an exception.